app/src/main/java/org/liberty/android/fantastischmemo/ui/QACardActivity.java

```diff
@@ -118,6 +119,8 @@
 
     private HintUtil hintUtil;
 
+    static HashSet<Integer> used = new HashSet<>();
```

Comment:
Why would you use a static variable here? Static means a global variable accessible by any class. Unless this is a global variable that can be accessed by any instance of the QACardActivity, this is very bad coding practice and should be remedy to.

Reply:
I wanted to make sure that it would work with all of the classes that use the HashSet. I changed it to have no modifier in order to make it accessible to all of the subclasses, in this case just HintUtil.